src/OrchardCore.Modules/OrchardCore.Search.Elasticsearch/Startup.cs

```diff
@@ -0,0 +1,261 @@
+                {
+                    var response = client.Ping();
+                    
+                    services.Configure<TemplateOptions>(o =>
+                    {
+                        o.MemberAccessStrategy.Register<SearchIndexViewModel>();
+                        o.MemberAccessStrategy.Register<SearchFormViewModel>();
+                        o.MemberAccessStrategy.Register<SearchResultsViewModel>();
+                    });
+
+                    services.AddElasticServices();
+                    services.AddSingleton<SearchProvider, ElasticSearchProvider>();
+                    services.AddScoped<IPermissionProvider, Permissions>();
+                    services.AddScoped<INavigationProvider, AdminMenu>();
+                    services.AddScoped<IDisplayDriver<ISite>, ElasticSettingsDisplayDriver>();
+                    services.AddScoped<IDisplayDriver<Query>, ElasticQueryDisplayDriver>();
+                    services.AddScoped<IContentTypePartDefinitionDisplayDriver, ContentTypePartIndexSettingsDisplayDriver>();
+                    services.AddScoped<IContentPartFieldDefinitionDisplayDriver, ContentPartFieldIndexSettingsDisplayDriver>();
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Elasticsearch is enabled but not active because the connection failed.");
+                }
+            }
+        }
+
+        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var options = serviceProvider.GetRequiredService<IOptions<ElasticConnectionOptions>>().Value;
+
+            if (!options.ConfigurationExists)
+            {
+                return;
+            }
+
+            var adminControllerName = typeof(AdminController).ControllerName();
+
+            routes.MapAreaControllerRoute(
+                name: "Elasticsearch.Index",
+                areaName: "OrchardCore.Search.Elasticsearch",
+                pattern: _adminOptions.AdminUrlPrefix + "/elasticsearch/Index",
+                defaults: new { controller = adminControllerName, action = nameof(AdminController.Index) }
+            );
+
+            routes.MapAreaControllerRoute(
+                name: "Elasticsearch.Delete",
+                areaName: "OrchardCore.Search.Elasticsearch",
+                pattern: _adminOptions.AdminUrlPrefix + "/elasticsearch/Delete/{id}",
+                defaults: new { controller = adminControllerName, action = nameof(AdminController.Delete) }
+            );
+
+            routes.MapAreaControllerRoute(
+                name: "Elasticsearch.Query",
+                areaName: "OrchardCore.Search.Elasticsearch",
+                pattern: _adminOptions.AdminUrlPrefix + "/elasticsearch/Query",
+                defaults: new { controller = adminControllerName, action = nameof(AdminController.Query) }
+            );
+
+            routes.MapAreaControllerRoute(
```

Comment:
Utilize https://github.com/OrchardCMS/OrchardCore/pull/12302 if that's merged before this one, to prevent the necessity of a manual rebuild, including doing the same with Elasticsearch. If the new docs also contain a reference to manual rebuilding (I didn't find one) then please remove that too.